src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -599,26 +577,28 @@ bool ColorControlServer::shouldExecuteIfOff(EndpointId endpoint, uint8_t optionM
  */
 void ColorControlServer::handleModeSwitch(EndpointId endpoint, uint8_t newColorMode)
 {
-    uint8_t oldColorMode = 0;
+    auto oldColorMode = ColorMode::kCurrentHueAndCurrentSaturation;
     Attributes::ColorMode::Get(endpoint, &oldColorMode);
 
     uint8_t colorModeTransition;
 
-    if (oldColorMode == newColorMode)
+    ColorControl::ColorMode newColorModeEnum                = static_cast<ColorControl::ColorMode>(newColorMode);
+    ColorControl::EnhancedColorMode newEnhanceColorModeEnum = static_cast<ColorControl::EnhancedColorMode>(newColorMode);
+    if (oldColorMode == newColorModeEnum)
     {
         return;
     }
 
-    Attributes::EnhancedColorMode::Set(endpoint, newColorMode);
-    if (newColorMode == ColorControl::EnhancedColorMode::kEnhancedCurrentHueAndCurrentSaturation)
+    Attributes::EnhancedColorMode::Set(endpoint, newEnhanceColorModeEnum);
+    if (newEnhanceColorModeEnum == ColorControl::EnhancedColorMode::kEnhancedCurrentHueAndCurrentSaturation)
     {
         // Transpose COLOR_MODE_EHSV to ColorControl::EnhancedColorMode::kCurrentHueAndCurrentSaturation after setting
         // EnhancedColorMode
-        newColorMode = ColorControl::EnhancedColorMode::kCurrentHueAndCurrentSaturation;
+        newEnhanceColorModeEnum = ColorControl::EnhancedColorMode::kCurrentHueAndCurrentSaturation;
     }
-    Attributes::ColorMode::Set(endpoint, newColorMode);
+    Attributes::ColorMode::Set(endpoint, newColorModeEnum);
 
-    colorModeTransition = static_cast<uint8_t>((newColorMode << 4) + oldColorMode);
+    colorModeTransition = static_cast<uint8_t>((to_underlying(newColorModeEnum) << 4) + to_underlying(oldColorMode));
```

Comment:
Yeah, this is clearly broken now too, with the above logic that never changes newColorModeEnum to be a legal non-enhanced value.  The fact that no tests fail on this is very bad.  @jmartinez-silabs